starknet_py/tests/e2e/docs/guide/test_using_cairo_serializer.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+from starkware.starknet.public.abi import get_selector_from_name
+from starknet_py.tests.e2e.utils import DevnetClientFactory
+
+
+@pytest.mark.asyncio
+async def test_using_cairo_serializer(run_devnet):
+    # pylint: disable=unused-variable, too-many-locals, import-outside-toplevel
+    # add to docs: start
+    from starknet_py.net.gateway_client import GatewayClient
+    from starknet_py.net.models import StarknetChainId
+    from starknet_py.contract import Contract
+    from starknet_py.net import AccountClient
+    from starknet_py.utils.data_transformer.data_transformer import CairoSerializer
+
+    # Code of the contract which emits an event
+    contract = """
+        %lang starknet
+        %builtins pedersen range_check
+        
+        from starkware.cairo.common.cairo_builtins import HashBuiltin
+        
+        @storage_var
+        func storage(key : felt) -> (value : felt):
+        end
+        
+        @event
+        func put_called(
+            key : felt, prev_value : felt, value : felt
+        ):
+        end
+        
+        @external
+        func put{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+                key : felt, value : felt):
+            let (prev_value) = storage.read(key)
+            put_called.emit(key=key, prev_value=prev_value, value=value)
+            storage.write(key, value)
+            return ()
+        end
+        """
+
+    net = "testnet"  # Can be "mainnet" or other custom net too
+    # add to docs: end
+
+    net = run_devnet
+    # add to docs: start
+
+    # Creates an account
+    client = await AccountClient.create_account(
+        client=GatewayClient(net=net, chain=StarknetChainId.TESTNET)
+    )
+    # add to docs: end
+
+    client = DevnetClientFactory(run_devnet).make_devnet_client()
+    # add to docs: start
+
+    # Deploys the contract
+    deployment_result = await Contract.deploy(client, compilation_source=contract)
+    await deployment_result.wait_for_acceptance()
+    contract = deployment_result.deployed_contract
+
+    # Invokes "put" function (which emits an event)
+    invoke_result = (
+        await contract.functions["put"].prepare(10, 20, max_fee=int(1e16)).invoke()
+    )
+
+    transaction_hash = invoke_result.hash
+    transaction_receipt = await client.get_transaction_receipt(transaction_hash)
+
+    # Takes events from transaction receipt
+    events = transaction_receipt.events
+
+    # Takes an abi of the event which data we want to serialize
+    emitted_event_abi = contract.data.abi[0]
```

Comment:
Could we have a more explicit example? I think this way it's a little bit unclear what are you actually assigning to emitted_event_abi, but maybe it's just me.

Reply:
You're right it's unclear. I can explicitly pass the event's abi here. It would be something like this:

emitted_event_abi = {'data': [{'name': 'key', 'type': 'felt'}, {'name': 'prev_value', 'type': 'felt'}, {'name': 'value', 'type': 'felt'}], 'keys': [], 'name': 'put_called', 'type': 'event'}

Is that ok?

Reply:
I think its better in this case.